pkg/storegateway/indexheader/reader_pool_test.go

```diff
@@ -151,15 +171,17 @@ func TestReaderPool_PersistLazyLoadedBlock(t *testing.T) {
 
 	err = snapshot.persist(tmpDir)
 	require.NoError(t, err)
-	require.Greater(t, snapshot.HeaderLastUsedTime[blockID], int64(0), "lazyLoadedBlocks snapshot must be set")
 
 	persistedFile := filepath.Join(tmpDir, lazyLoadedHeadersListFile)
 	persistedData, err := os.ReadFile(persistedFile)
 	require.NoError(t, err)
-	expectedSnapshot := &lazyLoadedHeadersSnapshot{}
-	err = json.Unmarshal(persistedData, expectedSnapshot)
-	require.NoError(t, err)
-	require.Equal(t, 1, len(expectedSnapshot.HeaderLastUsedTime), "should have one lazyLoaded index header")
+
+	var expected string
+	// we know that there is only one lazyReader, hence just use formatter to set the ULID and timestamp.
+	for r := range pool.lazyReaders {
+		expected = fmt.Sprintf(`{"header_last_used_time":{"%s":%d},"user_id":"anonymous"}`, r.blockID, r.usedAt.Load()/int64(time.Millisecond))
+	}
+	require.Equal(t, expected, string(persistedData))
```

Comment:
`JSONEq` is probably a better choice here, as it's resilient to whitespace and field order changes:

```suggestion
	require.JSONEq(t, expected, string(persistedData))
```